src/boolean_result.cpp

```diff
@@ -809,6 +810,40 @@ Manifold::Impl Boolean3::Result(OpType op) const {
   PRINT(n12 << " new verts from edgesP -> facesQ");
   PRINT(n21 << " new verts from facesP -> edgesQ");
 
+  const int numProp = std::max(inP_.NumProp(), inQ_.NumProp());
+  if (numProp > 0) {
+    // Ideally, switch over from triProperties vec to Halfedge.startProp - then
+    // the prop index will be available in the append edges functions below.
```

Comment:
@pca006132 I'm struggling to get this one finished - are you interested in taking a look? If you look at my last commit, I've added comments on the refactor I have planned. I think it's pretty straight forward, and I bet you'd optimize it better than I would. 

I ask because this is probably the last major PR to land before doing a release, right? We have a lot of good stuff now (mostly yours) and I don't want to delay it. Let me know what you think - and is my description clear at all?

Reply:
Sure, I can have a look at it later tonight.

Reply:
So this is kind of similar to what CreateProperties is doing, but you want to do it in parallel, and probably handle edges differently?

About the inclusion number array, should it be indexed by the vertex id, halfedge id, prop vert id or something?

Reply:
Yes, index by prop vert. Every prop vert has exactly one vert, so that makes things nice. The idea is to keep track of properties directly by index just like the verts, instead of needing to look anything up with maps or hash tables.